File: PlaybyPlayEFCore10/myapi/Seed.cs
using System;
using System.Linq;
using System.Collections.Generic;
using Microsoft.Extensions.DependencyInjection;
using EFCoreWebAPI;
using Newtonsoft.Json;

public static class Seeder {
  public static void Seedit(string jsonData, IServiceProvider serviceProvider) {
   
    List<WeatherEvent> events =
      JsonConvert.DeserializeObject<List<WeatherEvent>>(jsonData);
    using (var serviceScope = serviceProvider.GetRequiredService<IServiceScopeFactory>().CreateScope())
    {
      var context = serviceScope.ServiceProvider.GetService<WeatherContext>();
      if (!context.WeatherEvents.Any()) {
        context.AddRange(events);
        context.SaveChanges();
      }
    }
  }
}

File: PlaybyPlayEFCore10/FSharp/myapi/Seed.fs
module MyAPI.DataAccess.Seeder

open System
open System.Linq
open System.Collections.Generic
open Microsoft.Extensions.DependencyInjection
open MyAPI.Domain
open Newtonsoft.Json

let Seedit (serviceProvider:  IServiceProvider) jsonData =
    let events = JsonConvert.DeserializeObject<List<WeatherEvent>>(jsonData)

    use serviceScope = serviceProvider.GetRequiredService<IServiceScopeFactory>().CreateScope()
    let context = serviceScope.ServiceProvider.GetService<WeatherContext>()

    let inline addWeather w = context.Add(w) |> ignore
    
    if not(context.WeatherEvents.Any()) then
        Seq.iter addWeather events
        context.SaveChanges() 
        |> ignore
      